mp/src/game/client/momentum/ui/HUD/hud_stickybombcharge.cpp

```diff
@@ -57,9 +61,38 @@ CHudStickyCharge::CHudStickyCharge(const char *pElementName)
     LoadControlSettings("resource/ui/HudStickyCharge.res");
 }
 
+void CHudStickyCharge::FireGameEvent(IGameEvent* pEvent)
+{
+    auto *pLocal = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pLocal)
+        return; 
+        
+    const auto pRunEntData = pLocal->GetCurrentUIEntData();
+    if (!pRunEntData)
+        return;
+
+    if (FStrEq(pEvent->GetName(), "zone_enter") && pRunEntData->m_bIsInZone && pRunEntData->m_iCurrentZone == 1)
```

Comment:
These events have the zone number passed in under [the `"num"` integer](https://github.com/momentum-mod/game/blob/develop/mp/game/momentum/resource/modevents.res#L34), and the `zone_enter` event would imply that they're inside the zone now, so you can get rid of the need for the pRunEntData altogether here.

Reply:
Never noticed this, the speedometer doesnt use it either. I pushed a small commit to use that & fix up some code around the stage enter/exit stuffs.